src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size = 0;
+    private Node<K, V> nullKeyNode = null;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            putForNullKey(value);
+        } else {
+            int index = getIndex(key);
+            addOrUpdateNode(index, key, value);
+            resizeIfNeeded();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return nullKeyNode == null ? null : nullKeyNode.value;
+        }
+
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (node.key.equals(key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putForNullKey(V value) {
+        if (nullKeyNode == null) {
+            size++;
+        }
+        nullKeyNode = new Node<>(null, value, null);
```

Comment:
Don't use getters and setters in class Node. Directly access the fields of Node since it's an inner class and you have access to its private fields.